planning/behavior_path_planner/src/utils/goal_planner/default_fixed_goal_planner.cpp

```diff
@@ -38,12 +49,82 @@ BehaviorModuleOutput DefaultFixedGoalPlanner::plan(
   return output;
 }
 
+bool isInAnyLane(const lanelet::ConstLanelets & candidate_lanelets, const Point2d & point)
+{
+  for (const auto & ll : candidate_lanelets) {
+    if (boost::geometry::covered_by(point, ll.polygon2d().basicPolygon())) {
+      return true;
+    }
+  }
+  return false;
+}
+
+bool isAllPointsInAnyLane(
+  const PathWithLaneId & refined_path, const lanelet::ConstLanelets & candidate_lanelets)
+{
+  Point2d path_point_point2D;
+
+  for (size_t i = 0; i < refined_path.points.size(); ++i) {
+    const PathPointWithLaneId & path_point = refined_path.points[i];
+    path_point_point2D.x() = path_point.point.pose.position.x;
+    path_point_point2D.y() = path_point.point.pose.position.y;
+    std::cerr << "refined_path.points[" << i << "]:" << std::endl;
+    std::cerr << "X:" << path_point_point2D.x() << std::endl;
+    std::cerr << "Y:" << path_point_point2D.y() << std::endl;
+  }
+
+  for (size_t i = 0; i < refined_path.points.size(); ++i) {
+    const PathPointWithLaneId & path_point = refined_path.points[i];
+    path_point_point2D.x() = path_point.point.pose.position.x;
+    path_point_point2D.y() = path_point.point.pose.position.y;
+    bool is_point_in_any_lanelet = isInAnyLane(candidate_lanelets, path_point_point2D);
+    if (!is_point_in_any_lanelet) {
+      std::cerr << "INVALID POINT: refined_path.points[" << i << "]" << std::endl;
+      return false;  // at least one path_point falls outside any lanelet
+    }
+  }
+  return true;
+}
+
+lanelet::ConstLanelets extractLaneletsFromPath(
```

Comment:
@HansOersted 
Plase define this function as a member function of `DefaultFixedGoalPlanner` because it depends on the state of the class.

Reply:
modified .hpp and .cpp
member function made